### PR TITLE
Add Faker hook

### PR DESCRIPTION
**Pull request by arossert**
This adds the needed source files for the "Faker" module to enable the dynamic loading for the providers.

This will only work with the next `Faker` release after my changes in [#927](https://github.com/joke2k/faker/pull/927).

Is it possible to limit the supported version in the hook?

### CODE REVIEW

File: PyInstaller/hooks/hook-faker.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+# Distributed under the terms of the GNU General Public License with exception
+# for distributing bootloader.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+#-----------------------------------------------------------------------------
+
+from PyInstaller.utils.hooks import collect_submodules, collect_data_files
+
+hiddenimports = collect_submodules('faker.providers')
+datas = (collect_data_files('text_unidecode') +
```
Comment (cclauss) [inline PyInstaller/hooks/hook-faker.py]: Travis tests fail on this line because they want the plus (+) moved down to the beginning of the next line.

Comment (arossert) [reply PyInstaller/hooks/hook-faker.py]: @cclauss what is the reason for the failure this time?

**Comment by cclauss** [comment]
Please try putting the text __  # noqa: W503__ at the end of line 14.  This will tell flake8 to ignore this error on this line.  The base problem is https://github.com/pyinstaller/pyinstaller/blob/develop/setup.cfg#L69 because once you tell flake8 to ignore _any_ error it enables _both_ W503 and W504 which are contradictory.  This is the fallout of long discussions in the PyCQA https://github.com/PyCQA/pycodestyle/issues/498

**Comment by cclauss** [comment]
https://www.traviscistatus.com :-(

**Comment by arossert** [comment]
I have changed it to be in one line, will it resolve this?

**Comment by cclauss** [comment]
E501 @ 13:80 - line too long (109 > 79 characters)

**Comment by arossert** [comment]
@cclauss I have added the *# noqa: W503 * at the end of the line but the builds are still failing

**Comment by cclauss** [comment]
Problem is https://travis-ci.org/pyinstaller/pyinstaller/jobs/513020606#L1062

**Comment by arossert** [comment]
@cclauss I just can't get it to work, I have used the same syntax as [hook-pylint.py](https://github.com/pyinstaller/pyinstaller/blob/develop/PyInstaller/hooks/hook-pylint.py) but it still fails

**Comment by arossert** [comment]
@cclauss is there something else I can do to make it work?

File: PyInstaller/hooks/hook-faker.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+# The full license is in the file COPYING.txt, distributed with this software.
+#-----------------------------------------------------------------------------
+
+from PyInstaller.utils.hooks import collect_submodules, collect_data_files
+
+hiddenimports = collect_submodules('faker.providers')
+datas = (
+        collect_data_files('text_unidecode') +
+        collect_data_files('faker.providers', include_py_files=True)
+        )
```
Comment (cclauss) [inline PyInstaller/hooks/hook-faker.py]:
```
datas = (collect_data_files('text_unidecode') +
         collect_data_files('faker.providers',
                            include_py_files=True))  # noqa: W504
```

File: PyInstaller/hooks/hook-faker.py (hunk 1 above)
Comment (cclauss) [reply PyInstaller/hooks/hook-faker.py]: See that 🚫 on the last line below...  That means that file does not end with an empty line... let’s hope that flake8 does not complain about that...

**Comment by cclauss** [comment]
Try moving the comment up two lines...
```python
datas = (collect_data_files('text_unidecode') +  # noqa: W504
         collect_data_files('faker.providers',
                            include_py_files=True))
```

**Comment by arossert** [comment]
@cclauss half way there 😃 
Travis was successful.

**Comment by arossert** [comment]
@cclauss thanks a lot for all the help, hope that it will get merged and released soon

**Comment by arossert** [comment]
@cclauss do you know when this will be merged?

**Comment by cclauss** [comment]
Sorry but I am not a committer on this project.  If I was it woulda been merged already.

**Comment by arossert** [comment]
@bjones1 can this be merged? 
I need this for my project that is using PyInstaller.

**Comment by bjones1** [comment]
I merged this in f2e3bca73a88ffc63a9d8c45e70d38018ba223b8.

**Pull request closed without merging**